### ISSUE TITLE
How to make the fusion depend on depth and poses only?

### ISSUE THREAD

**Issue by cdb0y511** [weders/RoutedFusion#12]
Hi, I am working on evaluating routed fusion on the RGB-D dataset.
The first thing is to prepare input data, like depth, instincts, and extrinsics.
However, I find the test_fusion.py depends on "model.binvox.128.binvox", which states as ground truth grid. And it feeds to pipeline.py 

fuse(self,batch,database,device):
 line 145
 self._extractor.forward(frame,batch['extrinsics'],batch['intrinsics'], 
 volume['current'],volume['origin'],volume['resolution'],volume['weights'])

which is defined as:

    def forward(self, depth, extrinsics, intrinsics, tsdf_volume, origin, resolution, weights_volume=None):
        '''
        Computes the forward pass of extracting the rays/blocks and the corresponding coordinates

        :param depth: depth map with the values that define the center voxel of the ray/block
        :param extrinsics: camera extrinsics matrix for mapping
        :param intrinsics: camera intrinsics matrix for mapping
        :param volume_gt: groundtruth voxel volume
        :param volume_current: current state of reconstruction volume
        :param origin: origin of groundtruth volume in world coordinates
        :param resolution: resolution of voxel volume
        :return: values/voxels of groundtruth and current as well as at its coordinates and indices
        '''
So I am confused, how does it work on real data without a ground truth grid? 
Thanks for any advice.

**Comment by weders** [comment]
I assume you don't have ground-truth. You can just initialize a zero ground-truth grid (as you don't train on it) and process the reconstructed meshes/grids.

**Comment by cdb0y511** [comment]
Thanks for your reply,
Could you provide the script for real data input?
I would like to try your pipeline for real data like Figure 6,7,8 in your paper.
How can I set the depth scale parameter? or just 1000  = 1 meter for the input depth map?

**Comment by weders** [comment]
You can use the exact same [script](https://github.com/weders/RoutedFusion/blob/master/test_fusion.py). You just need to write a dataset class (similar to [this](https://github.com/weders/RoutedFusion/blob/master/dataset/shapenet.py)) with a get_grid(self, scene, truncation) method and a scenes property.

The input should be in metric scale and value 1 in the depth map corresponds to 1 meter.

**Comment by cdb0y511** [comment]
Thanks, I will try it later.
I think the zero ground-truth grid and the depth map scale is the reason why I get wrong output for depth map from the sensor.

**Comment by makeme-zgz** [comment]
@weders Should the pretrained fusion model work with different depth scale and depth map resolution? I'm quite confused with the relationship between the latent feature volume resolution vs the input depth map resolution. 

I think that the volume resolution should be larger than the depth map resolution since the ray that goes through a certain edge of the depth map needs to be considered. Please correct me if my understanding is wrong.

**Comment by ttzzj** [comment]
@cdb0y511 

Have you reproduced the routed fusion? Can I ask you a question? Paid is OK. Can you give me an email
